src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs

```diff
@@ -76,25 +84,17 @@ private void OnApplicationStopping()
             Logger.LogInformation("Application is shutting down...");
         }
 
-        private void OnProcessExit(object sender, EventArgs e)
+        private void OnCancelKeyPress(object sender, ConsoleCancelEventArgs e)
         {
-            ApplicationLifetime.StopApplication();
-            if (!_shutdownBlock.WaitOne(HostOptions.ShutdownTimeout))
-            {
-                Logger.LogInformation("Waiting for the host to be disposed. Ensure all 'IHost' instances are wrapped in 'using' blocks.");
-            }
-            _shutdownBlock.WaitOne();
-            // On Linux if the shutdown is triggered by SIGTERM then that's signaled with the 143 exit code.
-            // Suppress that since we shut down gracefully. https://github.com/dotnet/aspnetcore/issues/6526
-            System.Environment.ExitCode = 0;
+            e.Cancel = true;
+            OnExitSignal();
         }
 
-        private void OnCancelKeyPress(object sender, ConsoleCancelEventArgs e)
+        private void OnExitSignal()
         {
-            e.Cancel = true;
             ApplicationLifetime.StopApplication();
 
-            // Don't block in process shutdown for CTRL+C/SIGINT since we can set e.Cancel to true
+            // Don't block in process shutdown for CTRL+C/SIGINT/SIGTERM since we can set e.Cancel to true
             // we assume that application code will unwind once StopApplication signals the token
             _shutdownBlock.Set();
```

Comment:
It looks like I should be able to move the shutdownBlock to the notnetcoreapp file now. We don’t ever wait on it otherwise.